src/server/req_manager.c

```diff
@@ -2316,10 +2315,34 @@ mgr_node_set(struct batch_request *preq)
 		if (numnodes == 1)
 			break;	/* just the one vnode */
 		else if (preq->rq_ind.rq_manager.rq_objtype == MGR_OBJ_HOST) {
+			int imom;
+			int update_mom_only = 0;
+			unsigned long mstate;
+
 			/* next vnode under the Mom */
 			if (++momidx >= ((mom_svrinfo_t *)(pmom->mi_data))->msr_numvnds)
 				break;	/* all down */
 			pnode = ((mom_svrinfo_t *)(pmom->mi_data))->msr_children[momidx];
+			if ((strcmp(plist->al_name, ATTR_NODE_state) == 0) && (plist->al_op == INCR)) {
+				/* Marking nodes offline.  We should only mark the children vnodes
+				 * as offline if no other mom is up.
+				 */
+				if (pnode->nd_nummoms > 1) {
+					for (imom = 0; imom < pnode->nd_nummoms; ++imom) {
+						mstate = ((mom_svrinfo_t *)(pnode->nd_moms[imom]->mi_data))->msr_state;
+						if (((mstate & INUSE_DOWN) == 0) && ((mstate & INUSE_OFFLINE) == 0)) {
```

Comment:
Can you please OR both the states into one and then do the check?

Reply:
Okay.